src/components/charts/SwipeChart.tsx

```diff
@@ -12,33 +13,39 @@ import { useHandler } from '../../hooks/useHandler'
 import { formatDate } from '../../locales/intl'
 import { MinimalButton } from '../buttons/MinimalButton'
 import { FillLoader } from '../progress-indicators/FillLoader'
+import { showWarning } from '../services/AirshipInstance'
 import { Theme, useTheme } from '../services/ThemeContext'
 import { ReText } from '../text/ReText'
 import { EdgeText } from '../themed/EdgeText'
 
-type Kline = [
-  openTime: number,
-  open: string,
-  high: string,
-  low: string,
-  close: string,
-  volume: string,
-  closeTime: number,
-  quoteAssetVolume: string,
-  numberOfTrades: number,
-  takerBuyBaseAssetVolume: string,
-  takerBuyQuoteAssetVolume: string,
-  ignore: string
-]
-
 type Timespan = 'year' | 'month' | 'week' | 'day' | 'hour'
-
 type AlignOrigin = 'center' | 'right' | 'left'
 
+type CoinGeckoDataPair = number[]
+interface CoinGeckoMarketChartRange {
+  prices: CoinGeckoDataPair[]
+  market_caps: CoinGeckoDataPair[]
+  total_volumes: CoinGeckoDataPair[]
+}
+const asCoinGeckoDataPair = asArray(asNumber)
```

Comment:
If this is a fixed size array you can use [asTuple](https://cleaners.js.org/#/reference?id=astuple)